ui/src/app/machines/views/MachineDetails/MachineTests/MachineTestsDetails/MachineTestsDetails.tsx

```diff
@@ -37,6 +43,16 @@ const MachineTestsDetails = (): JSX.Element | null => {
     }
   }, [dispatch, scriptResults, loading, id]);
 
+  useEffect(() => {
+    if (!logs && result) {
+      ["combined", "stdout", "stderr", "result"].forEach((type) =>
+        dispatch(scriptResultActions.getLogs(result.id, type))
+      );
+    }
+  }, [dispatch, result, logs]);
+
+  const log = logs ? logs[parseInt(scriptResultId, 10)] : null;
```

Comment:
Instead of selecting all the logs and then find the log we need here should we instead implement a `getLog` selector?